### PR TITLE
[ADD] udes-stock: Added u_product_barcode_regex field

### PR DESCRIPTION
**Pull request by apatrick88**
Story/11931

Signed-off-by: Adam Patrick <adam.patrick@unipart.io>

### CODE REVIEW

File: addons/udes_stock/models/stock_warehouse.py [hunk 1]
```diff
@@ -78,6 +78,8 @@ def _domain_warehouse_picking_types(self):
                                          default='^PAL(\\d)+$')
     u_package_barcode_regex = fields.Char('Package Barcode Format',
                                           default='^PACK(\\d)+$')
+    u_product_barcode_regex = fields.Char('Product Barcode Replacement Regex',
```
Comment (williamsajr) [inline addons/udes_stock/models/stock_warehouse.py:81]: Could you update the README for this new field as well

Comment (apatrick88) [reply addons/udes_stock/models/stock_warehouse.py:81]: Will do

Comment (apatrick88) [reply addons/udes_stock/models/stock_warehouse.py:81]: Done

**Review by williamsajr — APPROVED** [review]

**Review by PeterC10 — APPROVED** [review]
LGTM

**Review by ws-mps — APPROVED** [review]

**Pull request merged**